### PR TITLE
Add user entity and migration

### PR DESCRIPTION
**Pull request by kmanser**
Creates a user entity and migration

### CODE REVIEW

**Comment by hmtranae** [comment]
user table should have property category_id that references categories(id)

File: src/main/resources/db/migration/V1.2__add_user_columns.sql [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ALTER TABLE users
 ADD COLUMN first_name VARCHAR(255) NOT NULL,
 ADD COLUMN last_name VARCHAR(255) NOT NULL,
-ADD FOREIGN KEY (category_id) REFERENCES Categories(category_id);
\ No newline at end of file
+ADD FOREIGN KEY (categories_id) REFERENCES categories(categories_id) NOT NULL;
```
Comment (hmtranae) [inline src/main/resources/db/migration/V1.2__add_user_columns.sql]: ADD FOREIGN KEY (category_id) REFERENCES categories(id) NOT NULL;

File: src/main/java/com/launchacademy/programmingswaglist/models/User.java [hunk 2]
```diff
@@ -43,4 +52,12 @@
...
   private Set<Role> roles;
 
+  @ManyToOne //Many categories to one user
+  @JoinColumn(name="category_id", nullable=false)
+  private Category category;
+
+  @OneToMany (mappedBy="users")//One user to many reviews
+  private List<Review> reviews = new ArrayList<Review>();
+
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/programmingswaglist/models/User.java]: please remove extra lines here

File: src/main/java/com/launchacademy/programmingswaglist/models/User.java [hunk 3]
```diff
@@ -43,4 +52,12 @@
   )
   private Set<Role> roles;
 
+  @ManyToOne //Many categories to one user
+  @JoinColumn(name="category_id", nullable=false)
+  private Category category;
+
+  @OneToMany (mappedBy="users")//One user to many reviews
+  private List<Review> reviews = new ArrayList<Review>();
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/programmingswaglist/models/User.java]: please remove comments prior to submitting PRs

Comment (kmanser) [reply src/main/java/com/launchacademy/programmingswaglist/models/User.java]: changes added, thank you!

File: src/main/java/com/launchacademy/programmingswaglist/models/User.java (hunk 2 above)
Comment (kmanser) [reply src/main/java/com/launchacademy/programmingswaglist/models/User.java]: changes added, thank you!

**Pull request merged**